src/v/kafka/server/usage_manager.cc

```diff
@@ -256,15 +257,48 @@ ss::future<> usage_manager::accounting_fiber::stop() {
     }
 }
 
-ss::future<> usage_manager::accounting_fiber::close_window() {
+void usage_manager::accounting_fiber::close_window() {
     const auto now = ss::lowres_system_clock::now();
-    _buckets[_current_window].u = co_await _um.map_reduce0(
-      [](usage_manager& um) { return um.sample(); },
-      _buckets[_current_window].u,
-      [](const usage& acc, const usage& x) { return acc + x; });
-    _buckets[_current_window].end = epoch_time_secs(now);
+    const auto now_ts = epoch_time_secs(now);
+    const auto before_close_idx = _current_window;
+    _buckets[before_close_idx].end = now_ts;
     _current_window = (_current_window + 1) % _buckets.size();
     _buckets[_current_window].reset(now);
+    /// The timer must progress so subsequent windows may be closed, async work
+    /// may hold that up for a significant amount of time, therefore async work
+    /// will be dispatched in the background and the result set written to the
+    /// correct bucket when it is eventually computed
+    ssx::spawn_with_gate(_gate, [this, before_close_idx, now_ts] {
+        return async_data_fetch(before_close_idx, now_ts)
+          .handle_exception([](std::exception_ptr eptr) {
+              vlog(
+                klog.debug,
+                "usage_manager async job exception encountered: {}",
+                eptr);
+          });
+    });
+}
+
+ss::future<> usage_manager::accounting_fiber::async_data_fetch(
+  size_t index, uint64_t close_ts) {
+    /// Method to write response to correct bucket, the index to write should be
+    /// \ref index, however if enough time has passed that window has been
+    /// overwritten, therefore the data can be omitted
+    const auto is_bucket_stale = [this, index, close_ts]() {
+        /// Check is simple, close_ts should be what it was when the window
+        /// closed, if it isn't, then the window is stale and was re-used, data
+        /// was overwritten.
+        return _buckets[index].end != close_ts;
```

Comment:
now that this runs asynchronously, where `index` is sampled in the past, can't i segfault the system by changing `.usage_num_windows.bind` config value strategically?

Reply:
No because when the config value changes, `stop()` is called on the accounting fiber, it is deallocated, and a new one is allocated and started and new configuration values passed to its constructor. 

Thats another reason why I went with this impl, at least for me it was easier to reason about these situations